### ISSUE TITLE
pg_dist_node uses wrong sequence names for the associated columns

### ISSUE THREAD

**Issue by onderkalaci** [citusdata/citus#1081]
This is a bug but nothing critical. I'm opening the issue to discuss solutions and whether we want to fix it before 6.1. or not.

With #798, we introduced `pg_dist_node` as a new metadata table. The definition of it:

```SQL
CREATE TABLE citus.pg_dist_node(
        nodeid int NOT NULL DEFAULT nextval('pg_dist_groupid_seq') PRIMARY KEY,
        groupid int NOT NULL DEFAULT nextval('pg_dist_node_nodeid_seq'),
        nodename text NOT NULL,
        nodeport int NOT NULL DEFAULT 5432,
        noderack text NOT NULL DEFAULT 'default',
        UNIQUE (nodename, nodeport)
);
```

As seen above, we assign `group_id` sequence to `nodeid` column. Similarly, 'nodeid_seq' to `groupid` column.
The output as expected:
```SQL
                            Table "pg_catalog.pg_dist_node"
   Column    |  Type   |                           Modifiers                           
-------------+---------+---------------------------------------------------------------
 nodeid      | integer | not null default nextval('pg_dist_groupid_seq'::regclass)
 groupid     | integer | not null default nextval('pg_dist_node_nodeid_seq'::regclass)
 nodename    | text    | not null
 nodeport    | integer | not null default 5432
 noderack    | text    | not null default 'default'::text
 hasmetadata | boolean | not null default false
Indexes:
    "pg_dist_node_pkey" PRIMARY KEY, btree (nodeid)
    "pg_dist_node_nodename_nodeport_key" UNIQUE CONSTRAINT, btree (nodename, nodeport)
Triggers:
    
```

**Comment by lithp** [comment]
Since 6.0 has already been released the proper fix is probably to, during UPGRADE, rename the sequences?

**Comment by marcocitus** [comment]
I think we need to alter the default column in the upgrade script. I would expect a rename to also "update" the default expressions.

**Comment by lithp** [comment]
> I think we need to alter the default column in the upgrade script.

This sounds like you're proposing the following?

```
ALTER TABLE pg_dist_node ALTER nodeid SET DEFAULT nextval('pg_dist_nodeid_seq');
ALTER TABLE pg_dist_node ALTER groupid SET DEFAULT nextval('pg_dist_groupid_seq');
```

This won't work because we've already released 6.0 and people have already used these sequences, so you're guaranteed to have a collision the next time you create a nodeid/groupid unless you happen to have an equal number of both.

> I would expect a rename to also "update" the default expressions.

That appears to be what happens:

```
brian=# CREATE TABLE test (a int, b int);
CREATE TABLE
brian=# CREATE SEQUENCE first_seq;
CREATE SEQUENCE
brian=# ALTER TABLE test ALTER a SET DEFAULT nextval('first_seq');
ALTER TABLE
brian=# \d test 
                    Table "public.test"
 Column |  Type   |               Modifiers                
--------+---------+----------------------------------------
 a      | integer | default nextval('first_seq'::regclass)
 b      | integer | 

brian=# ALTER SEQUENCE first_seq RENAME TO second_seq;
ALTER SEQUENCE
brian=# \d test 
                    Table "public.test"
 Column |  Type   |                Modifiers                
--------+---------+-----------------------------------------
 a      | integer | default nextval('second_seq'::regclass)
 b      | integer | 
```

We'll have to send one through a temporary name in order to swap the two of them, but I think the fix is just three rename commands.
